### DISCUSSION TITLE
Capture concentration at t-1

### DISCUSSION THREAD

**Discussion by krinaj** [nlmixrdevelopment/RxODE#386] [Q&A]
Hello Matt,

Is there a way to capture value for a compartment at previous time point, i.e., t-1 within ODE equation? I am trying to implement a model where I need to reassign a variable whether or not Conc > Conc at previous timepoint.

thanks,
Krina

**Comment by mattfidler** [comment]
Yes.  You can do this with RxODE 1.0+;  The release notes are here:

https://github.com/nlmixrdevelopment/RxODE/blob/master/NEWS.md



> Allow accessing different time-varying components of an input dataset for each individual with:

 >    lag(var, #)
 >   lead(var, #)
  >  first(var)
  >  last(var)
  >  diff(var)

They behave similar to their `R` equivalent functions.  

You can also use the new  sticky left hand sided of the solution;

 >  Allow sticky left-handed side of the equation; This means for an observation the left handed values are saved for the next observations and then reassigned to the last calculated value.

 >   This allows NONMEM-style of calculating parameters like tad:
```r
mod1 <-RxODE({
    KA=2.94E-01;
    CL=1.86E+01;
    V2=4.02E+01;
    Q=1.05E+01;
    V3=2.97E+02;
    Kin=1;
    Kout=1;
    EC50=200;
    C2 = centr/V2;
    C3 = peri/V3;
    d/dt(depot) =-KA*depot;
    d/dt(centr) = KA*depot - CL*C2 - Q*C2 + Q*C3;
    d/dt(peri)  =                    Q*C2 - Q*C3;
    d/dt(eff)  = Kin - Kout*(1-C2/(EC50+C2))*eff;
    if (!is.na(amt)){
        tdose <- time
    } else {
        tad <- time - tdose
    }
})
```

Hopefully this answers your question.

**Reply by mattfidler** [reply]
I moved this to a discussion forum, since it really isn't an issue.
